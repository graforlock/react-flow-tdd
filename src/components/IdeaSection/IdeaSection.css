.idea__list {
  display: flex;
  flex-direction: row;
  flex-flow: wrap;
  list-style: none;
  padding: 0;
}