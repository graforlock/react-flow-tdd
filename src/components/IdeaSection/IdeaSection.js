// @flow
import React, { Component } from 'react'
import { connect } from 'react-redux'
import './IdeaSection.css'

import { fetchCreateIdea } from '../../actions'
import type { IdeasFetchAction, IdeaRequest, IdeasStore } from '../../types'

import IdeaItem from '../IdeaItem'
import CreateIdea from '../CreateIdea'

type Props = {
  ideas: IdeasStore,
  message: ?string,
  fetchCreateIdea: (idea: IdeaRequest) => IdeasFetchAction
}

class IdeaSection extends Component<Props> {
  render () {
    const { ideas, message } = this.props

    return (
      <section className='idea__section'>
        <ul className='idea__list'>
          { ideas.map(idea => <IdeaItem key={idea.id} idea={idea} />) }
        </ul>
        { message && <span>{ message }</span> }
        <section className='idea__create'>
          <CreateIdea />
        </section>
      </section>
    )
  }
}

const mapStateToProps = ({ ideas, message }) => {
  return { ideas, message }
}

export default connect(mapStateToProps, {
  fetchCreateIdea
})(IdeaSection)
