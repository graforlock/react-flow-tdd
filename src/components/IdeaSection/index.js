import IdeaSection from './IdeaSection'

export default IdeaSection
