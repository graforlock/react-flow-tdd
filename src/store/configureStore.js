import { createStore, applyMiddleware, compose } from 'redux'
import createSagaMiddleware, { END } from 'redux-saga'
import rootReducer from '../reducers'

import type { ActionCreator, Store } from 'redux'
import type { SagaMiddlewareOptions, Task } from 'redux-saga'
import type { ReduxState } from '../reducers/initialState'
import persistState from 'redux-localstorage'

type ReduxStore = Store<ReduxState, ActionCreator<*>> & { runSaga: Task<*> }

const composeEnhancers = window.__REDUX_DEVTOOLS_EXTENSION_COMPOSE__ || compose
export default function configureStore (initialState: ReduxState): ReduxStore {
  const sagaMiddleware: SagaMiddlewareOptions<any> = createSagaMiddleware()
  const store = createStore(
    rootReducer,
    initialState,
    composeEnhancers(
      applyMiddleware(
        sagaMiddleware
      ),
      persistState()
    ),
    window.__REDUX_DEVTOOLS_EXTENSION__ && window.__REDUX_DEVTOOLS_EXTENSION__()
  )

  store.runSaga = sagaMiddleware.run
  store.close = () => store.dispatch(END)
  return store
}
