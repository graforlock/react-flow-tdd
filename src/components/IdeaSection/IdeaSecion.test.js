// @flow
import React from 'react'
import configureStore from 'redux-mock-store'
import { Provider } from 'react-redux'
import type { MockStore } from 'redux-mock-store'
import type { ShallowWrapper } from 'enzyme'
import { mount } from 'enzyme'

import IdeaSection from './IdeaSection'

const middlewares = []
const mockStore = configureStore(middlewares)

describe('IdeaSection', () => {
  const idea = {
    id: 1,
    title: 'Title',
    body: '... body ...',
    created_date: new Date()
  }
  const initialState = { ideas: [idea], editor: { writingMode: false } }
  let wrapper: ShallowWrapper
  let store: MockStore<any>

  beforeEach(() => {
    store = mockStore(initialState)
    wrapper = mount(<Provider store={store}><IdeaSection /></Provider>)
  })

  it('renders component and matches current snapshot', () => {
    expect(wrapper).toMatchSnapshot()
  })

  it('contains IdeaItems components', () => {
    expect(wrapper.find('IdeaItem').length).not.toBe(0)
  })
})
