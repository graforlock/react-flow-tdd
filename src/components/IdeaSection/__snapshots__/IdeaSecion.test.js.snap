// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`IdeaSection renders component and matches current snapshot 1`] = `
<Provider
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
>
  <Connect(IdeaSection)>
    <IdeaSection
      fetchCreateIdea={[Function]}
      ideas={
        Array [
          Object {
            "body": "... body ...",
            "created_date": 2018-02-27T13:15:42.954Z,
            "id": 1,
            "title": "Title",
          },
        ]
      }
    >
      <section
        className="idea__section"
      >
        <header>
          Idea app
        </header>
        <ul
          className="idea__list"
        >
          <Connect(IdeaItem)
            idea={
              Object {
                "body": "... body ...",
                "created_date": 2018-02-27T13:15:42.954Z,
                "id": 1,
                "title": "Title",
              }
            }
            key="1"
          >
            <IdeaItem
              fetchDeleteIdea={[Function]}
              idea={
                Object {
                  "body": "... body ...",
                  "created_date": 2018-02-27T13:15:42.954Z,
                  "id": 1,
                  "title": "Title",
                }
              }
            >
              <li
                className="idea__item"
              >
                <section
                  className="idea__item-content"
                >
                  <header
                    className="idea__header"
                  >
                    <Connect(TextEditor)
                      field="TITLE"
                      idea={
                        Object {
                          "body": "... body ...",
                          "created_date": 2018-02-27T13:15:42.954Z,
                          "id": 1,
                          "title": "Title",
                        }
                      }
                    >
                      <TextEditor
                        clearEditor={[Function]}
                        fetchUpdateIdea={[Function]}
                        field="TITLE"
                        idea={
                          Object {
                            "body": "... body ...",
                            "created_date": 2018-02-27T13:15:42.954Z,
                            "id": 1,
                            "title": "Title",
                          }
                        }
                        setEditorBody={[Function]}
                        setEditorTitle={[Function]}
                      >
                        <section
                          className="texteditor__container"
                        >
                          <h5
                            className="idea__title"
                            key=".0"
                            onClick={[Function]}
                          >
                            Title
                          </h5>
                          <div />
                        </section>
                      </TextEditor>
                    </Connect(TextEditor)>
                  </header>
                  <Connect(TextEditor)
                    field="BODY"
                    idea={
                      Object {
                        "body": "... body ...",
                        "created_date": 2018-02-27T13:15:42.954Z,
                        "id": 1,
                        "title": "Title",
                      }
                    }
                  >
                    <TextEditor
                      clearEditor={[Function]}
                      fetchUpdateIdea={[Function]}
                      field="BODY"
                      idea={
                        Object {
                          "body": "... body ...",
                          "created_date": 2018-02-27T13:15:42.954Z,
                          "id": 1,
                          "title": "Title",
                        }
                      }
                      setEditorBody={[Function]}
                      setEditorTitle={[Function]}
                    >
                      <section
                        className="texteditor__container"
                      >
                        <p
                          className="idea__body"
                          key=".0"
                          onClick={[Function]}
                        >
                          ... body ...
                        </p>
                        <div />
                      </section>
                    </TextEditor>
                  </Connect(TextEditor)>
                </section>
                <div
                  className="idea__delete-item"
                >
                  <div
                    className="idea__delete-button"
                    onClick={[Function]}
                  >
                    <img
                      alt="Delete icon"
                      className="idea__delete-icon"
                      src="trash-icon.svg"
                    />
                    <button
                      className="idea__delete-text"
                    >
                      Delete
                    </button>
                  </div>
                </div>
              </li>
            </IdeaItem>
          </Connect(IdeaItem)>
        </ul>
        <section
          className="idea__create"
        >
          <Connect(CreateIdea)>
            <CreateIdea
              disableEditorWriting={[Function]}
              enableEditorWriting={[Function]}
              fetchCreateIdea={[Function]}
              writingMode={false}
            >
              <button
                onClick={[Function]}
              >
                Add a new Idea
              </button>
            </CreateIdea>
          </Connect(CreateIdea)>
        </section>
      </section>
    </IdeaSection>
  </Connect(IdeaSection)>
</Provider>
`;
